src/storage/adaptive.rs

```diff
@@ -46,15 +49,23 @@ where
     T: StorageProvider,
 {
     type Reader = AdaptiveStorageReader<T::Reader>;
+    type Writer = AdaptiveStorageWriter<T::Writer>;
 
-    fn create_reader(&self, content_length: Option<u64>) -> io::Result<Self::Reader> {
-        if let Some(content_length) = content_length {
-            Ok(AdaptiveStorageReader::Unbounded(
-                self.inner.create_reader(Some(content_length))?,
-            ))
+    fn into_reader_writer(
```

Comment:
I think the wording here could be changed a bit, maybe we could call it `split` which would match [Tokio's API](https://docs.rs/tokio/latest/tokio/io/fn.split.html). Feels like kinda the same thing.

Reply:
In Tokio's case split makes sense as the input stream implements the same functionality as the read and write half do together. The [rust API guidelines](https://rust-lang.github.io/api-guidelines/naming.html) recommend a name with into for owned to owned type(s) conversions.  Given functionality is enabled/created here and it takes self as an argument I think that advise applies here. Do you agree?

Reply:
Fair enough, I agree with that. I'm good with this approach then.